### PR TITLE
♻️ Is3318/resend sms: refactoring (1/2)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

This PR does some refactoring before adding new resend entrypoint.

- ``api/specs/webserver``
   - ♻️ cleanup ``api/specs/webserver/openapi-auth.yaml``
   - 🔨 created ``api/specs/webserver/scripts/openapi_auth.py``  to auto generate ``openapi-auth.yaml`` specs.
   - ♻️ moved schemas from individual files to in-file ``components/schemas``  section (otherwise cannot use ``scripts/openapi-pydantic-models-generator.bash``)
- ``services/web/server`` - ``login`` plugin
    - ♻️ Splits ``/login/handlers.py ``
        - ``handlers_change``: handles for changing email/password etc
        - ``handlers_confirmation``:  handlers to confirm any action
        - ``handlers_registration``: handlers for user registration
        - Improved authentication error messages for 2FA
    - ♻️ Splits ``/login/utils.py`` and moves all emails functionality away into ``/login/utils_emails.py``

## Related issue/s

-  Refactoring before implementing #3318

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]

File: api/specs/webserver/openapi-auth.yaml [hunk 1]
```diff
@@ -373,3 +371,202 @@ components:
...
+        email:
+          type: string
+          #FIXME:  'error': {'logs': [], 'errors': [{'code': 'OpenAPISchemaError', 'message': 'Unsupported email format unmarshalling', 'resource': None, 'field': None}], 'status': 400}}
+          #format: email
+        password:
+          type: string
+          # TODO:   File "/home/crespo/devp/osparc-simcore/.venv/lib/python3.6/site-packages/openapi_core/schema/schemas/models.py", line 182, in _unmarshal_string
+          # formatter = self.STRING_FORMAT_CAST_CALLABLE_GETTER[schema_format]
+          # KeyError: <SchemaFormat.PASSWORD: 'password'>
+          #format: password
```
Comment (GitHK) [inline api/specs/webserver/openapi-auth.yaml:387]: Are these forgotten?

File: api/specs/webserver/scripts/openapi_auth.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+from _common import Error, Log
+from fastapi import FastAPI, status
+from models_library.generics import Envelope
+from pydantic import BaseModel, EmailStr, Field, SecretStr, confloat
+from simcore_service_webserver.login.api_keys_handlers import ApiKeyCreate, ApiKeyGet
+
+# TODO: move to simcore_service_webserver.login._models
+# TODO: how to ensure this is in sync with projects_ports_handlers.routes ??
+# this is the source of truth.
+
```
Comment (GitHK) [inline api/specs/webserver/scripts/openapi_auth.py:22]: I see some more todos here as well

File: services/web/server/src/simcore_service_webserver/login/handlers.py [hunk 3]
```diff
@@ -328,8 +98,8 @@ async def login(request: web.Request):
         if not user["phone"]:
             rsp = envelope_response(
                 {
-                    "code": "PHONE_NUMBER_REQUIRED",  # this string is used by the frontend
-                    "reason": "PHONE_NUMBER_REQUIRED",
+                    "code": LoginCode.PHONE_NUMBER_REQUIRED,  # this string is used by the frontend to show phone registration page
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/login/handlers.py:101]: I think you can remove the comment, since you added it to the enum

**Review by GitHK — APPROVED** [review]
:+1: just some questions below

File: api/specs/webserver/openapi-auth.yaml (hunk 1 above)
Comment (pcrespov) [reply api/specs/webserver/openapi-auth.yaml:387]: Those are pretty old TODO :-) I will remove them

File: api/specs/webserver/scripts/openapi_auth.py (hunk 2 above)
Comment (pcrespov) [reply api/specs/webserver/scripts/openapi_auth.py:22]: those i forgot indeed :-(

**Pull request merged**